dht_net.go

```diff
@@ -318,7 +318,7 @@ func (ms *messageSender) prep(ctx context.Context) error {
 		return nil
 	}
 
-	nstr, err := ms.dht.host.NewStream(ctx, ms.p, ms.dht.protocols...)
+	nstr, err := ms.dht.host.NewStream(ctx, ms.p, ms.dht.clientProtocols...)
```

Comment:
Please add doc explaining that we ONLY want to talk to peers who speak the "new" protocol.